plugins/defaultplugins/aclplugin/vppdump/dump_vppcalls.go

```diff
@@ -26,94 +26,252 @@ import (
 	"github.com/ligato/vpp-agent/plugins/defaultplugins/aclplugin/vppcalls"
 	acl_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/acl"
 	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/model/acl"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
 )
 
-// IPACLList defines the list of IP ACLs returned by the DumpIPAcl function
-type IPACLList struct {
-	Index      uint32               `json:"acl_index"`
+// ACLIdentifier contains fields for ACL index and Tag (used as a name in the configuration)
+type ACLIdentifier struct {
+	ACLIndex uint32 `json:"acl_index"`
+	Tag      string `json:"acl_tag"`
+}
+
+// ACLEntry is cumulative object with ACL identification and details with all ruleData and
+// interfaces belonging to the ACL
+type ACLEntry struct {
+	Identifier *ACLIdentifier
 	ACLDetails *acl.AccessLists_Acl `json:"acl_details"`
 }
 
-// DumpInterfaceAcls finds interface in VPP and returns its ACL configuration
-func DumpInterfaceAcls(log logging.Logger, swIndex uint32, vppChannel *govppapi.Channel,
-	timeLog measure.StopWatchEntry) (acl.AccessLists, error) {
+// ACLToInterface is definition of interface and all ACLs which are bound to
+// the interface either as ingress or egress
+type ACLToInterface struct {
+	SwIfIdx    uint32
+	IngressACL []uint32
+	EgressACL  []uint32
+}
 
-	log.Info("DumpInterfaceAcls")
-	// ACLInterfaceListDump time measurement
-	alAcls := acl.AccessLists{
-		Acl: []*acl.AccessLists_Acl{},
+// DumpACLs return a list of all configured ACLs including ruleData and interfaces
+func DumpACLs(log logging.Logger, swIfIndices ifaceidx.SwIfIndex, vppChannel *govppapi.Channel,
+	timeLog measure.StopWatchEntry) ([]*ACLEntry, error) {
+
+	var ACLs []*ACLEntry
+
+	// Get all IP and MACIP ruleData with particular ACL index
+	ruleData, err := DumpACLRules(log, vppChannel, timeLog)
+	if err != nil {
+		return nil, err
+	}
+	// Prepare separate list of all active ACL indices on the VPP
+	var indices []uint32
+	for identifier := range ruleData {
+		indices = append(indices, identifier.ACLIndex)
+	}
+
+	// Get all ACL indices with ingress and egress interfaces
+	interfaceData, err := DumpACLInterfaces(indices, swIfIndices, log, vppChannel, timeLog)
+	if err != nil {
+		return nil, err
 	}
 
+	// Build a list of ACL ruleData with ruleData, interfaces, index and tag (name)
+	for identifier, rules := range ruleData {
+		ACLs = append(ACLs, &ACLEntry{
+			Identifier: identifier,
+			ACLDetails: &acl.AccessLists_Acl{
+				Rules:      rules,
+				Interfaces: interfaceData[identifier.ACLIndex],
+			},
+		})
+	}
+
+	return ACLs, nil
+}
+
+// DumpACLRules returns all ruleData for every ACL
+func DumpACLRules(log logging.Logger, vppChannel *govppapi.Channel,
+	timeLog measure.StopWatchEntry) (map[*ACLIdentifier][]*acl.AccessLists_Acl_Rule, error) {
+	// rule map will be returned
+	rules := make(map[*ACLIdentifier][]*acl.AccessLists_Acl_Rule)
```

Comment:
I believe that using pointer to `ACLIdentifier` as key in map is not correct here. 
The map might contain two different values with keys that basically contain same field contents just different pointer or the map won't return the value for a different pointer key which basically contains same field contents as well. See this example for better explanation: https://play.golang.org/p/WCbYsyWKTc-